pkg/services/x509/x509_validator.go

```diff
@@ -0,0 +1,270 @@
+package x509
+
+import (
+	"bytes"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/lestrrat-go/jwx/jwa"
+	"github.com/lestrrat-go/jwx/jws"
+	"github.com/lestrrat-go/jwx/jwt"
+)
+
+type JwtX509Token struct {
+	chain  []*x509.Certificate
+	token  jwt.Token
+	raw    string
+	sigAlg jwa.SignatureAlgorithm
+}
+
+type JwtX509Validator struct {
+	roots              []*x509.Certificate
+	intermediates      []*x509.Certificate
+	allowedSigningAlgs []jwa.SignatureAlgorithm
+}
+
+// generalNames defines the asn1 data structure of the generalNames as defined in rfc5280#section-4.2.1.6
+type generalNames struct {
+	OtherName otherName `asn1:"tag:0"`
+}
+
+// otherName defines the asn1 data structure of the othername as defined in rfc5280#section-4.2.1.6
+type otherName struct {
+	OID   asn1.ObjectIdentifier
```

Comment:
do these need to be exported when the struct is unexported?

Reply:
Yes, otherwise the `asn1.Unmarshall` cant write to them.